src/Catty/DataModel/VariablesContainer/VariablesContainer.m

```diff
@@ -412,52 +412,46 @@ - (NSArray*)objectListsForObject:(SpriteObject*)spriteObject
     return lists;
 }
 
-- (SpriteObject*)spriteObjectForObjectVariable:(UserVariable*)userVariable
-{
-    if (!userVariable.isList)
-    {
-        for (NSUInteger index = 0; index < [self.objectVariableList count]; ++index) {
-            SpriteObject *spriteObject = [self.objectVariableList keyAtIndex:index];
-            NSMutableArray *userVariableList = [self.objectVariableList objectAtIndex:index];
-            for (UserVariable *userVariableToCompare in userVariableList) {
-                if (userVariableToCompare == userVariable) {
-                    return spriteObject;
-                }
+- (SpriteObject*)spriteObjectForObjectData:(id<UserDataProtocol>)userData
+{
+    
+    for (NSUInteger index = 0; index < [self.objectVariableList count]; ++index) {
+        SpriteObject *spriteObject = [self.objectVariableList keyAtIndex:index];
+        NSMutableArray *userVariableList = [self.objectVariableList objectAtIndex:index];
+        for (UserVariable *userVariableToCompare in userVariableList) {
+            if (userVariableToCompare == userData) {
+                return spriteObject;
             }
         }
     }
-    if (userVariable.isList)
-    {
-        for (NSUInteger index = 0; index < [self.objectListOfLists count]; ++index) {
-            SpriteObject *spriteObject = [self.objectListOfLists keyAtIndex:index];
-            NSMutableArray *userListOfLists = [self.objectListOfLists objectAtIndex:index];
-            for (UserVariable *userListToCompare in userListOfLists) {
-                if (userListToCompare == userVariable) {
-                    return spriteObject;
-                }
+    
+    for (NSUInteger index = 0; index < [self.objectListOfLists count]; ++index) {
+        SpriteObject *spriteObject = [self.objectListOfLists keyAtIndex:index];
+        NSMutableArray *userListOfLists = [self.objectListOfLists objectAtIndex:index];
+        for (UserList *userListToCompare in userListOfLists) {
+            if (userListToCompare == userData) {
+                return spriteObject;
             }
         }
     }
+    
     return nil;
 }
 
-- (BOOL)isProjectVariable: (UserVariable*)userVariable
+- (BOOL)isProjectData:(id<UserDataProtocol>)userData
 {
     for (UserVariable *userVariableToCompare in self.programVariableList) {
-        if ([userVariableToCompare.name isEqualToString:userVariable.name]) {
+        if ([userVariableToCompare.name isEqualToString:userData.name]) {
             return YES;
         }
     }
-    return NO;
-}
-
-- (BOOL)isProjectList: (UserVariable*)userList
-{
-    for (UserVariable *userListToCompare in self.programListOfLists) {
-        if ([userListToCompare.name isEqualToString:userList.name]) {
+    
+    for (UserList *userListToCompare in self.programListOfLists) {
```

Comment:
Please change to _UserVariable_ since this array still contains a list of variables.